Users/toml_config.rst

```diff
@@ -0,0 +1,286 @@
+user wide settings. Those settings will automatically be taken for all
+projects executed with that user.
+
+All settings specified here override only settings given by the system
+wide configuration file which has the lowest priority. The
+``system_coafile.toml`` must lie in the coala installation directory
+and is valid for everyone using this coala installation.
+
+It can be used to define the type of files you usually don't want to lint,
+like minified files (e.g. ``*.min.js``) and backup files (e.g. ``*.orig``)::
+
+    ignore = [ '**.min.js', '**.orig' ]
+
+Basic TOML concepts
+~~~~~~~~~~~~~~~~~~~~
+This part describes the basic TOML concepts required to write coala
+configuration files in TOML
+
+- TOML is case sensitive. So remember to not have duplicate sections/tables
+  or duplicate keys in same section.
+- key-value pairs are building blocks of a TOML document. Use key-value
+  pairs to specify rules for coala bears.
+- A table is a collection of key-value pairs. Use a table for specifying
+  a coala section.
+
+This is an example of a coala configuration file written in TOML ::
+
+    [cli]
+    bears = 'SpaceConsistencyBear'
+    files = 'src/*.c'
+    use_spaces = true
+
+    [invalidlinks]
+    enabled = false
+    files = [ '**/*.rst', 'README.rst']
+    ignore = 'venv/**'
+    bears = 'InvalidLinkBear'
+
+
+Here tables ``cli`` and ``invalidlinks`` are coala sections.
+The contents of the tables like ``bears``, ``files`` are rules
+that govern a section. To write coala configuration file you will
+be using TOML strings, booleans, integers and arrays as values.
+
+Section Inheritance
+~~~~~~~~~~~~~~~~~~~~
+coala supports section inheritance. You can define section inheritance
+by using the key ``inherits``.
+Extra values can be appended to an inherited setting using the ``appends`` key.
+
+Consider the following configuration file in TOML ::
+
+  [all]
+  enabled = true
+  overridable = 2
+  ignore = 'vendor1/'
+
+  [section1]
+  overridable = 3
+  inherits = 'all'
+  appends = 'ignore'
+  ignore = 'vendor2/'
+  other = 'some_value'
+
+  [section2]
+  overridable = 4
+  inherits = 'all'
+  ignore = 'vendor3/'
+  appends = 'ignore'
+  other = 'some_other_value'
+
+
+In the inherited sections above, ``appends`` key specifies that the value of
+``ignore`` in the derived sections must be appended with the value of
+``ignore`` key in the base section.
+
+This is the same file without section inheritance::
+
+  [all]
+  enabled = true
+  overridable = 2
+  ignore = 'vendor1/'
+
+  [section1]
+  enabled = true
+  overridable = 3
+  ignore = ['vendor1/', 'vendor2/']
+  other = 'some_value'
+
+  [section2]
+  enabled = true
+  overridable = 4
+  ignore = ['vendor1/', 'vendor3/']
+  other = 'some_other_value'
+
+
+Consider another example
+
+Config file in TOML
+
+::
+
+ [all]
+ a = 1
+ b = 2
+
+ [java]
+ c = 3
+ d = 4
+
+ [python]
+ p = 5
+ q = 6
+ inherits = [ 'all', 'java']
```

Comment:
What will happen when someone writes `all.a`, if nothing happens, mention that `inherits` can be a section name only